### ISSUE TITLE
[Feature Request] Allow to specify a parent node for Notification, Message, etc. at creation time instead of using document.body

### ISSUE THREAD

**Issue by Giwayume** [element-plus/element-plus#1743]
### Existing Component
Yes

### Component Name
Notification, Message

### Description
I'm writing an application that can be leveraged on other web pages, which means scoping the element-plus CSS so it doesn't interfere with the host application's styles. Because the CSS is scoped, components like Notification and Message will be unstyled if they're placed directly under the body instead of a specific node or shadow DOM. Because of this, I need to be able to specify the node the element is placed underneath, instead of it always defaulting to placing the component at document.body.

**Comment by JeremyWuuuuu** [comment]
Yeah. I think you've made your point. We can add APIs for the support of the request you asked.
And I think the request is quite legit, we will add it in the coming version, and I will keep you notified.

**Comment by JeremyWuuuuu** [comment]
The API should remain the same as `Vue` built-in `Teleport`'s API **to**, which would be default to 'body' and directly passed to `Teleport`.

**Comment by Giwayume** [comment]
Yeah, there may be other components I'm not thinking of that could benefit from this, too. I can kind of work around popover by using `:append-to-body="false"`, but there's a limitation to this in that I can't have a popover inside of another popover due to `overflow: hidden`.

**Comment by siryan2** [comment]
How is the status for this? 
I also looking for a way to not append the popover-container to the body and instead define a custom node.

**Comment by emojiiii** [comment]
please see [docs/appendTo](https://element-plus.gitee.io/en-US/component/message.html#options).
If you have any problems in using it, please let us know. @siryan2

**Comment by siryan2** [comment]
Thx for the answer. I will check it out and let you know if this also works with the popover component.

**Comment by Giwayume** [comment]
@emojiiii 
el-popover doesn't appear to have a passthrough to send the "append-to" prop to the el-tooltip.

**Comment by emojiiii** [comment]
> @emojiiii el-popover doesn't appear to have a passthrough to send the "append-to" prop to the el-tooltip.

you are right.

**Comment by FrontEndDog** [comment]
fixed by #4012
